core/app/models/spree/shipment.rb

```diff
@@ -262,6 +262,10 @@ def shipping_method
       selected_shipping_rate.try(:shipping_method) || shipping_rates.first.try(:shipping_method)
     end
 
+    def shipment_address
+      self.try(:address) || order.ship_address
```

Comment:
IMHO I'd prefer we did a migration to cleanup nil address data issues rather than work around them.